File: java/matrix/src/main/java/Matrix.java
import java.util.Arrays;

class Matrix {
    private final int[][] matrix;

    Matrix(String matrixAsString) {
        String[] lines = matrixAsString.split(System.lineSeparator());
        this.matrix = new int[lines.length][lines.length];
        for (int i = 0; i < lines.length; i++) {
            this.matrix[i] = Arrays.stream(lines[i].split(" ")).mapToInt(Integer::parseInt).toArray();
        }
    }

    int[] getRow(int rowNumber) {
        return matrix[rowNumber - 1];
    }

    int[] getColumn(int columnNumber) {
        int[] column = new int[this.matrix.length];
        for (int i = 0; i < this.matrix.length; i++) {
            column[i] = this.matrix[i][columnNumber - 1];
        }
        return column;
    }
}

File: kotlin/matrix/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    fun column(colNr: Int): List<Int> =
        matrixAsString.split("\n").map { line -> line.split(" ")[colNr - 1].toInt() }

    fun row(rowNr: Int): List<Int> =
        matrixAsString.split("\n")[rowNr - 1].split(" ").map { it.toInt() }

}
